common/lexical-model-types/build.sh

```diff
@@ -0,0 +1,144 @@
+#!/bin/bash
+#
+# Packages the lexical-model-types package.
+#
+
+# Exit on command failure and when using unset variables:
+set -eu
+
+# Include some helper functions from resources
+. ../../resources/shellHelperFunctions.sh
+EX_USAGE=64
+
+
+display_usage ( ) {
+  echo "Usage: $0 [-test] [-version version] [-tier tier]"
+  echo "       $0 -help"
+  echo
+  echo "  -help               displays this screen and exits"
+  echo "  -version version    sets the package version before building"
+  echo "  -test               runs tests"
+  echo "  -publish-to-npm     publishes the current version to the npm package index"
+  echo "  -tier tier          also sets the package version tier and npm tag (alpha, beta, stable) before building or publishing"
+  echo "                      If version has 4 components, only first three are used."
+}
+
+################################ Main script ################################
+
+run_tests=0
+install_dependencies=1
+publish_version=
+publish_tier=
+lastkey=
+should_publish=0
+npm_dist_tag=
+
+# Process command-line arguments
+while [[ $# -gt 0 ]] ; do
+  key="$1"
+  if [[ -z "$lastkey" ]]; then
+    case $key in
+      -help|-h)
+        display_usage
+        exit
+        ;;
+      -test)
+        run_tests=1
+        install_dependencies=0
+        ;;
+      -version)
+        lastkey=$key
+        ;;
+      -tier)
+        lastkey=$key
+        ;;
+      -publish-to-npm)
+        should_publish=1
+        ;;
+      *)
+        echo "$0: invalid option: $key"
+        display_usage
+        exit $EX_USAGE
+    esac
+  else
+    case $lastkey in
+      -version)
+        publish_version=$key
+        ;;
+      -tier)
+        publish_tier=$key
+        ;;
+      *)
+        # Should be impossible to reach ;-)
+        echo "$0: invalid option: $lastkey"
+        display_usage
+        exit $EX_USAGE
+    esac
+    lastkey=
+  fi
+  shift # past the processed argument
+done
+
+# Validate the publish_version
+if [ ! -z "$publish_version" ]; then
+  # Remove final component if more than 3 components passed
+  publish_version=` echo "$publish_version" | cut -d "." -f1,2,3 -`
+  [[ $publish_version =~ ^([0-9]+)\.([0-9]+)\.([0-9]+)$ ]] || fail "-version must be dotted numeric string, e.g. 1.2.3."
```

Comment:
You can use `assertValidVersionNbr()` from shellHelperFunctions.sh

Reply:
Careful here, this actually converts `12.0.1.0` to `12.0.1` as well as validating... not sure that `assertValidVersionNbr()` does that. Need that because the Windows build has the 4 part version number (another thing to tidy up!).